.github/workflows/build.yml

```diff
@@ -72,57 +64,39 @@ jobs:
         java: [ 8, 9, 10, 11, 12, 13, 14, 15, 16 ]
 
     steps:
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-          key: ${{ runner.os }}-gradle-${{ matrix.java }}-${{ hashFiles('**/*.gradle*') }}
-          restore-keys: ${{ runner.os }}-${{ matrix.java }}-gradle-
-
-      - uses: actions/checkout@v2
-
-      - name: Set up JDK ${{ matrix.java }}
-        uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-
       - name: Build and test
         if: startsWith(matrix.os, 'windows') == true
-        run: ./gradlew.bat build shadowJar proguard
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build shadowJar proguard
```

Comment:
https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.